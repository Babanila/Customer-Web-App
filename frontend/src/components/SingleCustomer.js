import React from 'react'
import { cx, css } from 'emotion'
import { utcDateConverter } from './Utils'

const tableRowDiv = css`
  display: table-row;
  background-color: #f6f6f6;
  &:hover {
    background-color: #ffe7e7;
  }
`

const tableCellDiv = css`
  display: table-cell;
  text-align: center;
  padding: 10px;
  border-bottom: 1px solid black;
`

function SingleCustomer({ customerData, onClick }) {
  const { customerID, name, lastContact, customerLifetimeValue } = customerData
  return (
    <div className={cx(tableRowDiv)} onClick={() => onClick(customerID)}>
      <div className={cx(tableCellDiv)}>{customerID}</div>
      <div className={cx(tableCellDiv)}>{name.first}</div>
      <div className={cx(tableCellDiv)}>{name.last}</div>
      <div className={cx(tableCellDiv)}>{utcDateConverter(lastContact)}</div>
      <div className={cx(tableCellDiv)}>{customerLifetimeValue}</div>
    </div>
  )
}

export default SingleCustomer
