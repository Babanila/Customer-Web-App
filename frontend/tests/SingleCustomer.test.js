import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import SingleCustomer from '../src/components/SingleCustomer'

describe('Check SingleCustomer Component', () => {
  let handleBack
  let sampleCustomerData

  beforeEach(() => {
    handleBack = jest.fn()
    sampleCustomerData = {
      customerID: 1,
      name: {
        first: 'Peter',
        last: 'Smith'
      },
      birthday: '1996-10-12',
      gender: 'm',
      lastContact: '2017-06-01T23:28:56.782Z',
      customerLifetimeValue: 191.12
    }
  })

  it('should renders without crashing', () => {
    const tree = renderer
      .create(<SingleCustomer customerData={sampleCustomerData} handleBack={handleBack} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correct number of element in SingleCustomer component', () => {
    const SingleCustomerWrapper = shallow(
      <SingleCustomer customerData={sampleCustomerData} handleBack={handleBack} />
    )

    const actual = SingleCustomerWrapper.find('div')
    expect(SingleCustomer).toBeDefined()
    expect(actual.exists).toBeTruthy()
    expect(actual.length).toEqual(6)
    expect(actual.at(0).text()).toContain('1PeterSmithThu, 01 Jun 2017 23:28:56 GMT191.12')
    expect(actual.at(0).childAt(0).text()).toEqual('1')
    expect(actual.at(0).childAt(2).text()).toEqual('Smith')
  })
})
