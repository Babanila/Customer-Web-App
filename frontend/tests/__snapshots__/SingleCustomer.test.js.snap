// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check SingleCustomer Component should renders without crashing 1`] = `
<div
  className="css-pe9o4c"
  onClick={[Function]}
>
  <div
    className="css-19d901a"
  >
    1
  </div>
  <div
    className="css-19d901a"
  >
    Peter
  </div>
  <div
    className="css-19d901a"
  >
    Smith
  </div>
  <div
    className="css-19d901a"
  >
    Thu, 01 Jun 2017 23:28:56 GMT
  </div>
  <div
    className="css-19d901a"
  >
    191.12
  </div>
</div>
`;
